pkg/virtctl/console/console.go

```diff
@@ -94,8 +94,17 @@ func (c *Console) Run(cmd *cobra.Command, args []string) error {
 	readStop := make(chan error)
 
 	go func() {
-		err := virtCli.VM(namespace).SerialConsole(vm, stdinReader, stdoutWriter)
-		resChan <- err
+		con, err := virtCli.VM(namespace).SerialConsole(vm)
+		if err != nil {
+			resChan <- err
+			return
+		}
+		defer con.Done()
```

Comment:
Looks to me like we can remove the `Done()` method from the interface and simply defer it in the `Stream()` call. I can't think of a situation where you would establish  the connection, check if there is no error, and in case that there is no error, not call `Stream()`. As far as I have seen in the k8s code that would also align with their implementation. What do you think?

Reply:
I think there is no need in the code for Done() - Which I added trying to be forward-looking, so removing it.